File: backend/src/main/java/com/github/ffremont/astrobook/domain/model/Picture.java
package com.github.ffremont.astrobook.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Builder(toBuilder = true)
@AllArgsConstructor
@NoArgsConstructor
@Getter
public class Picture {
    String id;
    String observationId;
    PictureState state;

    Object moonPhase;
    Weather weather;
    String instrument;
    String location;
    String camera;
    String corrRed;
    Float exposure;
    Integer gain;

    Integer stackCnt;
    List<String> tags;
    String constellation;
    String hash;
    String novaAstrometryReportUrl;
    Float ra;
    Float dec;
}

File: frontend/src/components/types/Picture.ts
export type Picture = Partial<{
    id: string
    moonPhase: string
    weather: string
    instrument: string
    location: string
    type: string
    name: string
    camera: string
    corrRed: string
    tags: string[]
    webTags: string[]
    exposure: number
    gain: number
    dateObs: string
    stackCnt: number
    constellation: string
    novaAstrometryReportUrl: string
}>